### PR TITLE
fix: Exclude trustarc url from auth header injection

### PR DESCRIPTION
**Pull request by charlesmulder**
## Summary by Sourcery

Extend the auth header injection logic by adding TrustArc analytics URLs to the exclusion list, exporting the isExcluded helper, and updating relevant tests and dependencies

Bug Fixes:
- Exclude TrustArc analytics endpoints from auth header injection

Enhancements:
- Add TrustArc analytics URL pattern to the exclusion list and export the isExcluded function

Build:
- Introduce @faker-js/faker as a devDependency
- Move pf-5-styles from devDependencies to dependencies

Tests:
- Add unit tests for isExcluded to cover excluded URLs, non-excluded URLs, and edge cases

### CODE REVIEW

File: src/utils/iqeEnablement.ts [hunk 1]
```diff
@@ -207,6 +210,7 @@ export function init(chromeStore: ReturnType<typeof createStore>, authRef: React
 
 const qe = {
   init,
+  isExcluded,
```
Comment (florkbr) [inline src/utils/iqeEnablement.ts:213]: @charlesmulder not blocking, but the name `isExcluded` is not very descriptive on _what_ exactly we are excluding and why (outside the context of this file). We may eventually want to update the name of this function to be a bit more clear to others if we expose it. 

Past that, @catastrophe-brandon could you forsee any reason this would cause issues with IQE? I don't think it would, but I just wanted to check.

File: src/utils/iqeEnablement.test.js [hunk 2]
```diff
@@ -18,3 +18,48 @@ describe('iqeEnablement', () => {
...
+    // random URLs
+    'https://example.com',
+    'https://api.redhat.com/some/endpoint',
+    'https://different-domain.com/analytics',
+    'https://api.example.com/upgrades_info',
+  ])('does not exclude %s', (url) => {
+    expect(iqeEnablement.isExcluded(url)).toBe(false);
+  });
+
+  // edge cases
```
Comment (florkbr) [inline src/utils/iqeEnablement.test.js:56]: 👍 for tests with the code!

File: src/utils/iqeEnablement.ts (hunk 1 above)
Comment (charlesmulder) [reply src/utils/iqeEnablement.ts:213]: @florkbr That's good thinking. Any name suggestions?

Comment (charlesmulder) [reply src/utils/iqeEnablement.ts:213]: @florkbr How about `isAuthHeaderDisabled`? It's a quick addition to this PR.
